llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -3743,7 +3743,7 @@ static SDValue getZeroVector(MVT VT, const X86Subtarget &Subtarget,
   SDValue Vec;
   if (!Subtarget.hasSSE2() && VT.is128BitVector()) {
     Vec = DAG.getConstantFP(+0.0, dl, MVT::v4f32);
-  } else if (VT.isFloatingPoint()) {
+  } else if (VT.isFloatingPoint() && VT.getVectorElementType() != MVT::bf16) {
```

Comment:
Why is this necessary? Are we missing bf16 handling somewhere?

Reply:
Because the scalar `bf16` is an illegal type even `vXbf16` are. We worked around it in BUILDVECTOR and should try to avoid build zero vector here.

Reply:
Could we test for isTypeLegal(VT.getVectorElementType()) instead?

Reply:
Done.